src/coredata.cs

```diff
@@ -1140,23 +1312,147 @@ public interface NSPersistentStore {
 		[Export ("type")]
 		string Type { get; }
 
-		[Export ("isReadOnly")]
+		[Export ("readOnly")]
 		bool ReadOnly { get; [Bind("setReadOnly:")] set; }
 
 		[Export ("metadata", ArgumentSemantic.Retain)]
-		NSDictionary Metadata { get; set; }
+		NSDictionary<NSString, NSObject> Metadata { get; set; }
 
 		[Export ("didAddToPersistentStoreCoordinator:")]
 		void DidAddToPersistentStoreCoordinator (NSPersistentStoreCoordinator coordinator);
 
 		[Export ("willRemoveFromPersistentStoreCoordinator:")]
-		void WillRemoveFromPersistentStoreCoordinator (NSPersistentStoreCoordinator coordinator);
+		void WillRemoveFromPersistentStoreCoordinator ([NullAllowed] NSPersistentStoreCoordinator coordinator);
 
 		[Since (5,0)]
 		[Field ("NSPersistentStoreSaveConflictsErrorKey")]
 		NSString SaveConflictsErrorKey { get; }
 
 	}
+	
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(NSObject))]
+	public interface NSPersistentStoreDescription : NSCopying
+	{
+		// +(instancetype _Nonnull)persistentStoreDescriptionWithURL:(NSURL * _Nonnull)URL;
+		[Static]
+		[Export ("persistentStoreDescriptionWithURL:")]
+		NSPersistentStoreDescription PersistentStoreDescriptionWithUrl (NSUrl Url);
```

Comment:
a better name would be `GetPersistentStoreDescription`, i.e. a prefix/verb + removal of the suffix which is covered by the parameter(s)